openpype/hosts/maya/plugins/load/load_reference.py

```diff
@@ -168,3 +179,47 @@ def _post_process_rig(self, name, namespace, context, options):
                 options={"useSelection": True},
                 data={"dependencies": dependency}
             )
+
+    def _lock_camera_transforms(self, nodes):
+        cameras = cmds.ls(nodes, type="camera")
+
+        # Check the Maya version, lockTransform has been introduced since
+        # Maya 2016.5 Ext 2
+        version = int(cmds.about(version=True))
+        if version >= 2016:
+            for camera in cameras:
+                cmds.camera(camera, edit=True, lockTransform=True)
+        else:
+            self.log.warning("This version of Maya does not support locking of"
+                             " transforms of cameras.")
+
+    @staticmethod
+    def _update_cameras(container):
+        # Get the modelPanels that used the old camera
+        members = get_container_members(container)
+        old_cameras = cmds.ls(members, type="camera", long=True)
+        update_panels = []
+        for panel in cmds.getPanel(type="modelPanel"):
+            cam = cmds.ls(cmds.modelPanel(panel, query=True, camera=True),
+                          long=True)
+
+            # Often but not always maya returns the transform from the
+            # modelPanel as opposed to the camera shape, so we convert it
+            # to explicitly be the camera shape
+            if cmds.nodeType(cam) != "camera":
+                cam = cmds.listRelatives(cam,
+                                         children=True,
+                                         fullPath=True,
+                                         type="camera")[0]
+            if cam in old_cameras:
+                update_panels.append(panel)
+        return update_panels
+
+    @staticmethod
+    def _update_model_panels(container, update_panels):
+        # Update the modelPanels to contain the new camera
+        members = get_container_members(container)
```

Comment:
Let's opt out early here if there are no panels to avoid the extra queries below.

```suggestion
        if not update_panels:
            return
        members = get_container_members(container)
```